fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -162,6 +165,10 @@ public class LuceneRecordCursor implements BaseCursor<IndexEntry> {
         this.pageSize = pageSize;
         this.executorService = executorService;
         this.storedFields = storedFields;
+        this.storedFieldsToReturn = Sets.newHashSet(LuceneIndexMaintainer.PRIMARY_KEY_FIELD_NAME);
+        if (this.storedFields != null) { // Only return StoredValues required for primary key and to support query.
```

Comment:
Trying to understand the role of the `storedFields` argument: Running queries with various `SELECT` fields (or `.*`) resulted in this parameter unchanged: it always contains 13 elements, 11 of which are `null` and the remaining 2 are `msgstate_uid` and `msgstate_flags`.

Reply:
So I guess my question is what should be the intended impact of this optimization?

Reply:
The optimization will not materialize the stored values that are not-needed based on the query.  We need primary key for each value but if you had a field for example that stored another value, it is not needed to be retrieved for the user unless the query user needs to return it.